pkg/controllers/dnspolicy/dnspolicy_healthchecks.go

```diff
@@ -122,52 +123,58 @@ func (r *DNSPolicyReconciler) expectedHealthCheckProbesForGateway(ctx context.Co
 		interval = *dnsPolicy.Spec.HealthCheck.Interval
 	}
 
-	for _, address := range gw.Status.Addresses {
-		matches := strings.Split(address.Value, "/")
-		if len(matches) != 2 {
-			log.V(3).Info(fmt.Sprintf("%s cannot be processed. expected <clsutername>/<hostname/ipaddress>", address.Value))
-			return nil
+	gatewayWrapper := utils.NewGatewayWrapper(gw.Gateway)
+	if err := gatewayWrapper.Validate(); err != nil {
+		return nil
+	}
+
+	clusterGatewayAddresses := gatewayWrapper.GetClusterGatewayAddresses()
+
+	for _, listener := range gw.Spec.Listeners {
+
+		//skip wildcard listeners
+		if strings.Contains(string(*listener.Hostname), "*") {
+			continue
 		}
 
-		for _, listener := range gw.Spec.Listeners {
-			if strings.Contains(string(*listener.Hostname), "*") {
-				continue
-			}
+		port := dnsPolicy.Spec.HealthCheck.Port
+		if port == nil {
+			listenerPort := int(listener.Port)
+			port = &listenerPort
+		}
 
-			port := dnsPolicy.Spec.HealthCheck.Port
-			if port == nil {
-				listenerPort := int(listener.Port)
-				port = &listenerPort
-			}
+		// handle protocol being nil
+		var protocol string
+		if dnsPolicy.Spec.HealthCheck.Protocol == nil {
+			protocol = string(listener.Protocol)
+		} else {
+			protocol = string(*dnsPolicy.Spec.HealthCheck.Protocol)
+		}
 
-			// handle protocol being nil
-			var protocol string
-			if dnsPolicy.Spec.HealthCheck.Protocol == nil {
-				protocol = string(listener.Protocol)
-			} else {
-				protocol = string(*dnsPolicy.Spec.HealthCheck.Protocol)
-			}
-			log.V(1).Info("reconcileHealthCheckProbes: adding health check for target", "target", address.Value)
-			healthCheck := &v1alpha1.DNSHealthCheckProbe{
-				ObjectMeta: metav1.ObjectMeta{
-					Name:      dnsHealthCheckProbeName(matches[1], gw.Name, string(listener.Name)),
-					Namespace: gw.Namespace,
-					Labels:    commonDNSRecordLabels(client.ObjectKeyFromObject(gw), client.ObjectKeyFromObject(dnsPolicy)),
-				},
-				Spec: v1alpha1.DNSHealthCheckProbeSpec{
-					Port:                     *port,
-					Host:                     string(*listener.Hostname),
-					Address:                  matches[1],
-					Path:                     dnsPolicy.Spec.HealthCheck.Endpoint,
-					Protocol:                 v1alpha1.HealthProtocol(protocol),
-					Interval:                 interval,
-					AdditionalHeadersRef:     dnsPolicy.Spec.HealthCheck.AdditionalHeadersRef,
-					FailureThreshold:         dnsPolicy.Spec.HealthCheck.FailureThreshold,
-					ExpectedResponses:        dnsPolicy.Spec.HealthCheck.ExpectedResponses,
-					AllowInsecureCertificate: dnsPolicy.Spec.HealthCheck.AllowInsecureCertificates,
-				},
+		for _, addresses := range clusterGatewayAddresses {
+			for _, address := range addresses {
```

Comment:
so what happens with hostname here?

Reply:
they're honoured, the logic in the gatewaywrapper class just looks at the type string - to see if it is a Kuadrant style value or regular.